File: {{ cookiecutter.repo_name }}/scripts/01_process_data.jl
"""01 - Process raw data into Arrow (exploratory, linear Julia script)."""

using DataFrames, CSV, Arrow
using Logging

include("../src/config.jl")

const INPUT_PATH = joinpath(RAW_DATA_DIR, "dataset.csv")
const OUTPUT_PATH = joinpath(PROCESSED_DATA_DIR, "dataset.arrow")

@info "Loading raw data from $INPUT_PATH"

if !isfile(INPUT_PATH)
    @warn "$INPUT_PATH not found, creating dummy data"
    mkpath(dirname(INPUT_PATH))
    df_dummy = DataFrame(x=0:99, y=[i^2 for i in 0:99])
    CSV.write(INPUT_PATH, df_dummy)
end

df = CSV.read(INPUT_PATH, DataFrame)
@info "Loaded $(nrow(df)) rows"

# --- Edit here: cleaning / filtering ---
# df = dropmissing(df)

@info "Saving to $OUTPUT_PATH"
mkpath(dirname(OUTPUT_PATH))
Arrow.write(OUTPUT_PATH, df)
@info "Done. Saved $(nrow(df)) rows to $OUTPUT_PATH"

File: {{ cookiecutter.repo_name }}/scripts/01_process_data.py
"""01 - Process raw data into Parquet (exploratory, linear script)."""

from loguru import logger
import pandas as pd

from {{ cookiecutter.module_name }}.config import PROCESSED_DATA_DIR, RAW_DATA_DIR

INPUT_PATH = RAW_DATA_DIR / "dataset.csv"
OUTPUT_PATH = PROCESSED_DATA_DIR / "dataset.parquet"

logger.info(f"Loading raw data from {INPUT_PATH}")
# For demo, create dummy data if file missing
if not INPUT_PATH.exists():
    logger.warning(f"{INPUT_PATH} not found, creating dummy data for demo")
    INPUT_PATH.parent.mkdir(parents=True, exist_ok=True)
    pd.DataFrame({"x": range(100), "y": [i**2 for i in range(100)]}).to_csv(
        INPUT_PATH, index=False
    )

df = pd.read_csv(INPUT_PATH)
logger.info(f"Loaded {len(df)} rows")

# --- Edit here: cleaning / filtering ---
# df = df.dropna()

logger.info(f"Saving to {OUTPUT_PATH}")
OUTPUT_PATH.parent.mkdir(parents=True, exist_ok=True)
df.to_parquet(OUTPUT_PATH, compression="snappy")
logger.success(f"Done. Saved {len(df)} rows to {OUTPUT_PATH}")
